### PR TITLE
Added locks that were missing in remoteEndpoint APIs.

### PR DESCRIPTION
**Pull request by gkvijay**
Change 8382c204 moved locking to the methods, except these new ones.

### CODE REVIEW

File: netplugin/plugin/netplugin.go [hunk 1]
```diff
@@ -151,11 +151,15 @@ func (p *NetPlugin) DeleteEndpoint(id string) error {
 
 // CreateRemoteEndpoint creates an endpoint for a given ID.
 func (p *NetPlugin) CreateRemoteEndpoint(id string) error {
+	p.Lock()
```
Comment (gaurav-dalvi) [inline netplugin/plugin/netplugin.go]: Looks good to me.

**Comment by dseevr** [comment]
@gaurav-dalvi Good catch on `AddPolicyRule`/`DelPolicyRule`.  `FetchNetwork` just returns an error (intentionally), so nothing needed there, and I think `Init` only gets called once.

One improvement that could be made to the file is to have the `Get*` and `Inspect*` functions use `RLock()` and `RUnlock()`, but that is a microoptimization

**Comment by unclejack** [comment]
The use of RLock and RUnlock would be a good idea. This would allow us to improve performance a bit.

**Comment by rchirakk** [comment]
LGTM

File: netplugin/plugin/netplugin.go [hunk 2]
```diff
@@ -151,11 +151,15 @@ func (p *NetPlugin) DeleteEndpoint(id string) error {
...
+	defer p.Unlock()
 	return p.NetworkDriver.CreateRemoteEndpoint(id)
 }
 
 // DeleteRemoteEndpoint destroys an endpoint for an ID.
 func (p *NetPlugin) DeleteRemoteEndpoint(id string) error {
+	p.Lock()
```
Comment (abhinandanpb) [inline netplugin/plugin/netplugin.go]: Isnt the lock acquired here also ?  https://github.com/contiv/netplugin/blob/b6719ea52b1ee96cd13016a868ada555bfe545eb/netplugin/agent/state_event.go#L279

File: netplugin/plugin/netplugin.go [hunk 3]
```diff
@@ -338,10 +342,14 @@ func (p *NetPlugin) DelSvcSpec(svcName string, spec *core.ServiceSpec) {
 
 // AddPolicyRule creates a policy rule
 func (p *NetPlugin) AddPolicyRule(id string) error {
+	p.Lock()
```
Comment (abhinandanpb) [inline netplugin/plugin/netplugin.go]: same here ? 
https://github.com/contiv/netplugin/blob/b6719ea52b1ee96cd13016a868ada555bfe545eb/netplugin/agent/state_event.go#L506

Comment (gkvijay) [inline netplugin/plugin/netplugin.go]: @abhinandanpb PTAL

**Pull request merged**